torch/_inductor/graph.py

```diff
@@ -1285,6 +1285,19 @@ def materialize(x):
             else:
                 real_inputs = [materialize(x) for x in V.real_inputs]
 
+            if self.mutated_inputs:
+                mutated_input_idxs = [
+                    idx
+                    for idx, name in enumerate(self.graph_inputs)
+                    if name in self.mutated_inputs
+                    and isinstance(real_inputs[idx], torch.Tensor)
+                ]
+                for idx in mutated_input_idxs:
+                    # clone mutated Tensor inputs to avoid mutating the them
+                    # in the first pass of the CPP wrapper compilation, as this
+                    # will lead to double mutation otherwise
```

Comment:
Hmm, I have a silly question - what does double mutation mean here? Thanks!

Reply:
It's misleading in the comment, let me rephrase it. But what I mean with "double mutation" is summarized in the [previous comment](https://github.com/pytorch/pytorch/pull/123316#issuecomment-2035996966). Please let me know if that clarifies.

Reply:
I've extended the comment, hope it reads more clear now.

Reply:
I see. Thanks for the clarification.